### PR TITLE
chore: add golangci-lint configuration

### PR DESCRIPTION
**Pull request by AlekSi**
Adapt it from Talos repo.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    - godot
+    - golint
+    - govet
+    - lll
+    - misspell
+    - nlreturn
+    - predeclared
+    - revive
+    - staticcheck
+    - stylecheck
```
Comment (AlekSi) [inline .golangci.yml:136]: Fix send PRs for enabling those linters

**Review by smira — DISMISSED** [review]
:fire:

**Comment by smira** [comment]
restarted the build: https://ci.dev.talos-systems.io/talos-systems/sidero/1230

**Review by smira — APPROVED** [review]

**Pull request merged**